src/browser-provider/browser.js

```diff
@@ -0,0 +1,60 @@
+import viewportSizes from 'viewportsizes';
+import uuid from 'uuid';
+import { assignIn } from 'lodash';
+
+
+export default class Browser {
+    constructor (provider, alias, options) {
+        this.provider = provider;
+        this.alias    = alias;
+        this.handle   = null;
+        this.options  = options;
+        this.uuid     = uuid.v4();
+    }
+
+    get id () {
+        return this.uuid;
+    }
+
+    async open (startPage) {
+        var descriptor = await this.provider.getDescriptorDefaultValues(this.alias);
```

Comment:
I guess it's not ready for review yet - we've decided to significantly change code (see discussion above)